esp/src/src-react/hooks/metrics.ts

```diff
@@ -2,96 +2,218 @@ import * as React from "react";
+        sql: `\
+SELECT type, name, CostExecute, TimeElapsed, id
+    FROM metrics`,
+        layout: undefined,
+        showTimeline: true
+    },
+    Graphs: {
+        scopeTypes: ["graph", "subgraph"],
+        properties: ["CostExecute", "TimeElapsed"],
+        ignoreGlobalStoreOutEdges: true,
+        subgraphTpl: "%id% - %TimeElapsed%",
+        activityTpl: "%Label%",
+        edgeTpl: "%Label%\n%NumRowsProcessed%\n%SkewMinRowsProcessed% / %SkewMaxRowsProcessed%",
+        sql: `\
+SELECT type, name, CostExecute, TimeElapsed, id
+    FROM metrics
+    WHERE type = 'graph' OR type = 'subgraph'`,
+        layout: undefined,
+        showTimeline: true
+    },
+    Activities: {
+        scopeTypes: ["activity"],
+        properties: ["TimeMaxLocalExecute", "TimeMaxTotalExecute"],
+        ignoreGlobalStoreOutEdges: true,
+        subgraphTpl: "%id% - %TimeElapsed%",
+        activityTpl: "%Label%",
+        edgeTpl: "%Label%\n%NumRowsProcessed%\n%SkewMinRowsProcessed% / %SkewMaxRowsProcessed%",
+        sql: `\
+SELECT type, name, TimeLocalExecute, TimeTotalExecute, id
+    FROM metrics
+    WHERE type = 'activity'`,
+        layout: undefined,
+        showTimeline: true
+    },
+    Peaks: {
+        scopeTypes: ["subgraph"],
+        properties: ["NodeMaxPeakMemory", "NodeMaxPeakRowMemory", "NodeMinPeakMemory", "NodeMinPeakRowMemory", "SizeAvgPeakMemory", "SizeAvgPeakRowMemory", "SizeDeltaPeakMemory", "SizeDeltaPeakRowMemory", "SizeMaxPeakMemory", "SizeMaxPeakRowMemory", "SizeMinPeakMemory", "SizeMinPeakRowMemory", "SizePeakMemory", "SizeStdDevPeakMemory", "SizeStdDevPeakRowMemory", "SkewMaxPeakMemory", "SkewMaxPeakRowMemory", "SkewMinPeakMemory", "SkewMinPeakRowMemory"],
+        ignoreGlobalStoreOutEdges: true,
+        subgraphTpl: "%id% - %TimeElapsed%",
+        activityTpl: "%Label%",
+        edgeTpl: "%Label%\n%NumRowsProcessed%\n%SkewMinRowsProcessed% / %SkewMaxRowsProcessed%",
+        sql: `\
+SELECT type, name, NodeMaxPeakMemory, NodeMaxPeakRowMemory, NodeMinPeakMemory, NodeMinPeakRowMemory, SizeAvgPeakMemory, SizeAvgPeakRowMemory, SizeDeltaPeakMemory, SizeDeltaPeakRowMemory, SizeMaxPeakMemory, SizeMaxPeakRowMemory, SizeMinPeakMemory, SizeMinPeakRowMemory, SizePeakMemory, SizeStdDevPeakMemory, SizeStdDevPeakRowMemory, SkewMaxPeakMemory, SkewMaxPeakRowMemory, SkewMinPeakMemory, SkewMinPeakRowMemory, id
+    FROM metrics
+    WHERE type = 'subgraph'`,
+        layout: undefined,
+        showTimeline: true
+    }
 };
 
-const options: MetricsOptions = { ...defaults };
+// const _defaultUserViews = JSON.stringify({ viewId: _viewId.value, views: _views.value });
 
-function checkLayout(options: MetricsOptions): boolean {
-    if (options?.layout && !options?.layout?.["main"]) {
-        delete options.layout;
-    }
-    return !!options?.layout;
+export function clone<T>(_: T): T {
+    return JSON.parse(JSON.stringify(_));
+}
+
+// function checkLayout(options: MetricsView): boolean {
```

Comment:
a bit of commented out code here and that _defaultUserViews a few lines up

Reply:
Actually want to call this...